src/wallet/scriptpubkeyman.cpp

```diff
@@ -1510,7 +1510,9 @@ bool DescriptorScriptPubKeyMan::GetNewDestination(const OutputType type, CTxDest
     {
         LOCK(cs_desc_man);
         assert(m_wallet_descriptor.descriptor->IsSingleType()); // This is a combo descriptor which should not be an active descriptor
-        if (type != m_address_type) {
+        Optional<OutputType> desc_addr_type = m_wallet_descriptor.descriptor->GetOutputType();
+        assert(desc_addr_type);
```

Comment:
just curious, when would this assert?

Reply:
If the descriptor was an `addr()`, `raw()`, or `combo()` descriptor. But it shouldn't be possible to reach this point with those descriptors.